### PR TITLE
Enables multi-file ansible hook in linchpin 

### PR DESCRIPTION
**Pull request by samvarankashyap**
fixes: #379 #748 
Summary:
Fixes Ansible hooks action_manager
Added integration test for multiple playbook runs for dummy provider
@p3ck

### CODE REVIEW

**Comment by p3ck** [comment]
Looks good.  Can we change the hooks to create some file which you can check for?  That way we can be sure the hooks actually ran.

File: docs/source/examples/workspaces/dummy-multi-file-hook/linchpin.conf [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+# The name of the state which occurs after (pre) teardown (destroy)
+#postdestroy = postdestroy
+
+# The name of the state which occurs after (post) inventory is generated (inv)
+#postinv = inventory
+#
+#[hookflags]
+#no_hooks = False
+#ignore_failed_hooks = False
+
```
Comment (herlo) [inline docs/source/examples/workspaces/dummy-multi-file-hook/linchpin.conf]: Unless this file differs from the one in ../linchpin.conf, why not make it a symlink. It works fine with fetch and the testing afaik.

**Comment by p3ck** [comment]
@samvarankashyap How do you know your second hook ran?  I want these tests to be able to verify that your hooks are running.  We shouldn't have to do manually review the tests to know if it worked.  Unless I'm missing something.

**Comment by samvarankashyap** [comment]
@p3ck Updated the hook test create a file and verify it with a cat command. 
@herlo : Removed unnecessary  linchpin.conf

File: config/Dockerfiles/tests.d/dummy/05-dummy-multi-file-hook [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+if [ -e /tmp/dummy.hosts ]; then
+    rm /tmp/dummy.hosts
+fi
+
+# testing no-hooks flag
+echo "Testing multi file hook"
+linchpin -w "${WORKSPACE_PATH}" -p "${PINFILE}" -v up "${TARGET}"
+
+cat /tmp/ex_hook.txt
```
Comment (p3ck) [inline config/Dockerfiles/tests.d/dummy/05-dummy-multi-file-hook]: We're getting closer...  Can you actually check the output of the file to verify it contains the data that should have been written out by the second hook?  And if it doesn't exit non-zero.

**Comment by samvarankashyap** [comment]
@p3ck Updated as per your suggestions

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**